src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -254,6 +257,26 @@ def app_post_deploy_hooks(self) -> Optional[List[ApplicationPostDeployHook]]:
     def debug_mode(self) -> bool:
         return self.na_project.debug_mode
 
+    @contextmanager
+    def use_warehouse(self, new_wh: Optional[str]):
+        """
+        Switches to a different warehouse for a while, then switches back.
+        This is a no-op if the requested warehouse is already active, or the requested warehouse is not available.
```

Comment:
If the requested warehouse is not available, shouldn't we throw an error?

Reply:
I will clarify the comment, but the distinction here is:
1. If the CLI is able to determine the app warehouse or a warehouse from the connection context, then use that, in that order.
2. If nothing can be found, then error out as usual.